### PR TITLE
Added function `create_offset_polygons_with_holes_2`

### PR DESCRIPTION
**Pull request by romanarust**
Added function `create_offset_polygons_with_holes_2` along with some code refactorings on `straight_skeleton_2.cpp`.

### CODE REVIEW

File: src/compas_cgal/straight_skeleton_2.py [hunk 1]
```diff
@@ -155,6 +155,63 @@ def create_offset_polygons_2(points, offset) -> list[Polygon]:
...
+    ----------
+    points : list of point coordinates or :class:`compas.geometry.Polygon`
+        The points of the polygon.
+    holes : list of list of point coordinates or list of :class:`compas.geometry.Polygon`
+        The holes of the polygon.
+    offset : float
+        The offset distance. If negative, the offset is outside the polygon, otherwise inside.
+
+    Returns
+    -------
```
Comment (tomvanmele) [inline src/compas_cgal/straight_skeleton_2.py:171]: remove

File: src/compas_cgal/straight_skeleton_2.py [hunk 2]
```diff
@@ -155,6 +155,63 @@ def create_offset_polygons_2(points, offset) -> list[Polygon]:
...
+    list of tuple of (:class:`Polygon`, list[:class:`Polygon`])
+        The polygons with holes.
+
+    Raises
+    ------
+    ValueError
+        If the normal of the polygon is not [0, 0, 1].
```
Comment (tomvanmele) [inline src/compas_cgal/straight_skeleton_2.py:181]: should it be a unit normal, or vertical up?

File: src/compas_cgal/straight_skeleton_2.py [hunk 3]
```diff
@@ -155,6 +155,63 @@ def create_offset_polygons_2(points, offset) -> list[Polygon]:
...
+    H = []
+    for i, hole in enumerate(holes):
+        points = list(hole)
+        normal_hole = normal_polygon(points, True)
+        if not TOL.is_allclose(normal_hole, [0, 0, -1]):
+            raise ValueError("The normal of the hole should be [0, 0, -1]. The normal of the provided {}-th hole is {}".format(i, normal_hole))
+        hole = np.asarray(points, dtype=np.float64)
+        H.append(hole)
+
+    offset = float(offset)
```
Comment (tomvanmele) [inline src/compas_cgal/straight_skeleton_2.py:199]: why the conversion?

Comment (tomvanmele) [reply src/compas_cgal/straight_skeleton_2.py:199]: in any case, would do this at the beginning of the algorithm. process all input params (if necessary) at the start...

File: src/compas_cgal/straight_skeleton_2.py [hunk 4]
```diff
@@ -155,6 +155,63 @@ def create_offset_polygons_2(points, offset) -> list[Polygon]:
...
+    if offset < 0:  # outside
+        offset_polygons = straight_skeleton_2.create_offset_polygons_2_outer_with_holes(V, H, abs(offset))
+    else:  # inside
+        offset_polygons = straight_skeleton_2.create_offset_polygons_2_inner_with_holes(V, H, offset)
```
Comment (tomvanmele) [inline src/compas_cgal/straight_skeleton_2.py:203]: are these algorithms so different that you need two separate ones? can the differentiation between inside and outside not simply be made at the C++ level?

File: src/compas_cgal/straight_skeleton_2.py [hunk 5]
```diff
@@ -155,6 +155,63 @@ def create_offset_polygons_2(points, offset) -> list[Polygon]:
     return [Polygon(points.tolist()) for points in offset_polygons]
 
 
+def create_offset_polygons_with_holes_2(points, holes, offset) -> list[Tuple[Polygon, list[Polygon]]]:
```
[the remark this answers is not in the document]
Comment (romanarust) [reply src/compas_cgal/straight_skeleton_2.py:158]: right, i've added it to the docstrings. I've use "_2" as the cgal functions are named like that as well

File: src/compas_cgal/straight_skeleton_2.py (hunk 2 above)
Comment (romanarust) [reply src/compas_cgal/straight_skeleton_2.py]: should be vertical up, i've adjusted the docstrings

File: src/compas_cgal/straight_skeleton_2.py [hunk 6]
```diff
@@ -155,6 +155,63 @@ def create_offset_polygons_2(points, offset) -> list[Polygon]:
...
+    """
+    points = list(points)
+    normal = normal_polygon(points, True)
+    if not TOL.is_allclose(normal, [0, 0, 1]):
+        raise ValueError("The normal of the polygon should be [0, 0, 1]. The normal of the provided polygon is {}".format(normal))
+    V = np.asarray(points, dtype=np.float64)
+
+    H = []
+    for i, hole in enumerate(holes):
+        points = list(hole)
```
[the remark this answers is not in the document]
Comment (romanarust) [reply src/compas_cgal/straight_skeleton_2.py]: right, not needed

File: src/compas_cgal/straight_skeleton_2.py (hunk 3 above)
Comment (romanarust) [reply src/compas_cgal/straight_skeleton_2.py]: you are right, it is not needed

File: src/compas_cgal/straight_skeleton_2.py (hunk 4 above)
Comment (romanarust) [reply src/compas_cgal/straight_skeleton_2.py:199]: yep, i could do it on c++ level as well, but I've already implemented it like that for the other functions in the same way. 
I guess it does not make a difference if the "if" is on python or c++ side

**Review by tomvanmele — APPROVED** [review]
LGTM

**Pull request merged**